File: pkg/client/network_test.go
// Copyright IBM Corp. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package client

import (
	"testing"

	"github.com/stretchr/testify/require"
)

func AssertNewTestNetwork(t *testing.T, networkName string, options ...ConnectOption) *Network {
	gateway := AssertNewTestGateway(t, options...)
	return gateway.GetNetwork(networkName)
}

func TestNetwork(t *testing.T) {
	t.Run("GetContract returns correctly named Contract", func(t *testing.T) {
		chaincodeName := "chaincode"
		network := AssertNewTestNetwork(t, "network")

		contract := network.GetContract(chaincodeName)

		require.NotNil(t, contract)
		require.Equal(t, chaincodeName, contract.ChaincodeName(), "chaincode name")
		require.Equal(t, "", contract.ContractName(), "contract name")
	})

	t.Run("GetContractWithName returns correctly named Contract", func(t *testing.T) {
		chaincodeName := "chaincode"
		contractName := "contract"
		network := AssertNewTestNetwork(t, "network")

		contract := network.GetContractWithName(chaincodeName, contractName)

		require.NotNil(t, contract)
		require.Equal(t, chaincodeName, contract.ChaincodeName(), "chaincode name")
		require.Equal(t, contractName, contract.ContractName(), "contract name")
	})
}

File: node/src/network.test.ts
/*
 * Copyright 2020 IBM All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import * as grpc from '@grpc/grpc-js';
import { connect, ConnectOptions } from './gateway';
import { Identity } from './identity/identity';
import { Network } from './network';

describe('Network', () => {
    let network: Network;
    let client: grpc.Client;

    beforeEach(() => {
        const identity: Identity = {
            mspId: 'MSP_ID',
            credentials: Buffer.from('CERTIFICATE'),
        };
        client = new grpc.Client('example.org:1337', grpc.credentials.createInsecure());
        const options: ConnectOptions = {
            identity,
            client,
        };

        const gateway = connect(options);
        network = gateway.getNetwork('CHANNEL_NAME');
    });

    describe('getContract', () => {
        it('returns correctly named default contract', () => {
            const contract = network.getContract('CHAINCODE_NAME');

            expect(contract.getChaincodeName()).toBe('CHAINCODE_NAME');
            expect(contract.getContractName()).toBeUndefined();
        });

        it('returns correctly named non-default contract', () => {
            const contract = network.getContract('CHAINCODE_NAME', 'CONTRACT_NAME');

            expect(contract.getChaincodeName()).toBe('CHAINCODE_NAME');
            expect(contract.getContractName()).toBe('CONTRACT_NAME');
        });
    });
});
